BraintreeCore/src/main/java/com/braintreepayments/api/core/AnalyticsEventRepository.kt

```diff
@@ -0,0 +1,40 @@
+package com.braintreepayments.api.core
+
+/**
+ * This class is responsible for holding analytic events in memory that will be batched with other events.
+ */
+internal class AnalyticsEventRepository {
+
+    private val events: MutableList<AnalyticsEvent> = mutableListOf()
+
+    /**
+     * Adds an event to the internal event buffer.
+     */
+    fun addEvent(event: AnalyticsEvent) {
+        events.add(event)
+    }
+
+    /**
+     * Flushes the current event buffer and returns the accumulated events.
+     *
+     * This function atomically clears the internal `events` list and returns a new list containing all the events that
+     * were previously in the buffer.
+     *
+     * @return A new list containing all events that were previously in the buffer.
+     */
+    fun flushAndReturnEvents(): List<AnalyticsEvent> {
+        synchronized(events) {
```

Comment:
Since this is synchronized should we be calling the analytics send method on a new thread each time?

Reply:
I believe we're calling `AnalyticsClient.sendEvent()` on the main thread in every instance, so having this synchronized block is probably overkill. But just wanted to add some safety if we end up sending events on different threads.

Reply:
We should be okay! This was all happening on the main thread previously. I was originally planning on introducing Coroutines for sending events, but saw we have a `BraintreeHttpClient.post()` that takes a callback and executes the call on a background thread. So I just went down that route to reduce the scope of the PR. I still think we can make future enhancements migrating to Coroutines.